client/asset/btc/btc.go

```diff
@@ -1799,13 +1799,8 @@ func (btc *ExchangeWallet) FindRedemption(ctx context.Context, coinID dex.Bytes)
 		contractHash: dexbtc.ExtractScriptHash(pkScript),
 	}
 
-	btc.findRedemptionMtx.Lock()
-	oldRedemption := btc.findRedemptionQueue[outPt]
-	btc.findRedemptionQueue[outPt] = req
-	btc.findRedemptionMtx.Unlock()
-
-	if oldRedemption != nil {
-		oldRedemption.fail("Duplicate FindRedemption request received. Aborting the old request.")
+	if err := btc.queueFindRedemptionRequest(req); err != nil {
+		return exitError("queueFindRedemptionRequest error for redemption %s: %v", outPt, err)
 	}
```

Comment:
This change is in response to https://github.com/decred/dcrdex/pull/1089#discussion_r684511117, but I'm not sure if it's better this way. The question is who should see the error, the previously waiting goroutine, or the new one.